addons/html_editor/static/src/others/dynamic_placeholder/dynamic_placeholder.js

```diff
@@ -0,0 +1,92 @@
+import { _t } from "@web/core/l10n/translation";
+import { Plugin } from "@html_editor/plugin";
+
+export class DynamicPlaceholderPlugin extends Plugin {
+    static name = "dynamicplaceholder";
+    static dependencies = ["selection", "history", "dom"];
+    static resources = (p) => ({
+        powerboxCategory: { id: "marketing_tools", name: _t("Marketing Tools"), sequence: 60 },
+        powerboxCommands: {
+            name: _t("Dynamic Placeholder"),
+            description: _t("Insert personalized content"),
+            category: "marketing_tools",
+            fontawesome: "fa-magic",
+            action(dispatch) {
+                dispatch("OPEN_DYNAMIC_PLACEHOLDER");
+            },
+        },
+    });
+    setup() {
+        if (!this.config.dynamicPlaceholder || this.config.dynamicPlaceholderModelReferenceField) {
+            throw new Error("Invalid Dynamic Placeholder Config");
+        }
+        this.dynamicPlaceholder = this.config.dynamicPlaceholder;
+        this.dynamicPlaceholderModelReferenceField =
+            this.config.dynamicPlaceholderModelReferenceField;
+    }
+
+    handleCommand(command) {
+        switch (command) {
+            case "OPEN_DYNAMIC_PLACEHOLDER": {
+                this.openDialog();
+
+                this.eventListenerHash = this.addDomListener(
```

Comment:
No need to save a hash if we use the map strategy (see comment for `addons/html_editor/static/src/plugin.js`)